DuckDuckGo/DownloadsList.swift

```diff
@@ -0,0 +1,187 @@
+    var body: some View {
+        NavigationView {
+            listOrEmptyState
+                .navigationBarTitle(Text(UserText.downloadsScreenTitle), displayMode: .inline)
+                .navigationBarItems(trailing: doneButton)
+        }
+        .navigationViewStyle(.stack)
+    }
+    
+    private var doneButton: some View {
+        Button(action: { presentationMode.wrappedValue.dismiss() },
+               label: { Text(UserText.navigationTitleDone).foregroundColor(.barButton).bold() })
+            .opacity(editMode == .inactive ? 1.0 : 0.0)
+    }
+    
+    @ViewBuilder
+    private var listOrEmptyState: some View {
+        if viewModel.sections.isEmpty {
+            emptyState
+        } else {
+            listWithBottomToolbar
+        }
+    }
+    
+    private var emptyState: some View {
+        VStack {
+            Spacer()
+                .frame(height: 32)
+            Text(UserText.emptyDownloads)
+                .font(Font(uiFont: Const.Font.emptyState))
+                .foregroundColor(.emptyState)
+            Spacer()
+        }
+        .frame(maxWidth: .infinity)
+        .background(Color.background)
+        .edgesIgnoringSafeArea(.bottom)
+    }
+    
+    @ViewBuilder
+    private var listWithBottomToolbar: some View {
+        if #available(iOS 14.0, *) {
+            list.toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Spacer()
+                    EditButton().environment(\.editMode, $editMode)
+                        .foregroundColor(.barButton)
+                }
+            }
+        } else {
+            // Due to no proper toolbar support in SwiftUI for iOS 13
+            VStack {
+                list
+                HStack {
+                    Spacer()
+                    EditButton().environment(\.editMode, $editMode)
+                        .foregroundColor(.barButton)
+                }
+                .padding()
+            }
+        }
+    }
+    
+    private var list: some View {
+        List {
+            ForEach(viewModel.sections) { section in
+                Section(header: Text(section.header)) {
+                    ForEach(section.rows) { row in
+                        if let row = row as? OngoingDownloadRowViewModel {
+                            OngoingDownloadRow(rowModel: row,
+                                               cancelButtonAction: { self.isCancelDownloadAlertPresented = true })
+                                .alert(isPresented: $isCancelDownloadAlertPresented) { makeCancelDownloadAlert(for: row) }
+                                .deleteDisabled(true)
+                        } else if let row = row as? CompleteDownloadRowViewModel {
+                            CompleteDownloadRow(rowModel: row,
+                                                shareButtonAction: { buttonFrame in
+                                share(row, from: buttonFrame)
+                            })
+                        }
```

Comment:
Small suggestion: Maybe move this if/let else/if to a separated @ViewBuilder function receiving a `DownloadsListRowViewModel` as parameter?  I think it would be easier to read considering that this var has some nesting.

So it could look something like this:
```
                    ForEach(section.rows) { row in
                         downloadRow(row)
                    }
```